src/payment/strategies/laybuy/laybuy-payment-strategy.ts

```diff
@@ -0,0 +1,79 @@
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import * as paymentStatusTypes from '../../payment-status-types';
+import PaymentStrategy from '../payment-strategy';
+
+export default class LaybuyPaymentStrategy implements PaymentStrategy {
+
+    constructor(
+        private _store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _formPoster: FormPoster
+    ) {}
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+        const paymentData = payment && payment.paymentData;
+
+        if (!payment || !paymentData) {
+            throw new PaymentArgumentInvalidError(['payment.paymentData']);
+        }
+
+        return this._store.dispatch(this._orderActionCreator.submitOrder(order, options))
+            .then(() =>
+                this._store.dispatch(this._paymentActionCreator.submitPayment({ ...payment, paymentData }))
+            )
+            .catch(error => {
+                if (!(error instanceof RequestError) || !some(error.body.errors, { code: 'insert error here' })) {
+                    return new Promise(() => {
+                        this._formPoster.postForm('https://sandbox-api.laybuy.com/order/create', {
+                            headers: {
+                                Accept: 'application/json',
+                                'Content-Type': 'application/json',
+                                authorization: 'Basic cGF5bWVudC1pbnRlZ3JhdGlvbnMrbnpAYmlnY29tbWVyY2UuY29tOk9UTXlOME09',
+                            },
+                        });
+                    });
+                }
+
+                // return Promise.reject('error');
+                return new Promise(() => {
+                    this._formPoster.postForm('https://sandbox-api.laybuy.com/order/create', {
+                        headers: {
+                            Accept: 'application/json',
+                            'Content-Type': 'application/json',
+                            authorization: 'Basic cGF5bWVudC1pbnRlZ3JhdGlvbnMrbnpAYmlnY29tbWVyY2UuY29tOk9UTXlOME09',
+                        },
+                    });
+                });
+            });
+    }
+
+    finalize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const state = this._store.getState();
+        const order = state.order.getOrder();
+
+        if (order && state.payment.getPaymentStatus() === paymentStatusTypes.FINALIZE) {
+            return this._store.dispatch(this._orderActionCreator.finalizeOrder(order.orderId, options));
+        }
+
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
```

Comment:
In case of Laybuy as part of our technical design, this method will not be used.

You can only add finalize method as: 
```suggestion
    finalize(): Promise<InternalCheckoutSelectors> {
        return Promise.reject(new OrderFinalizationNotRequiredError());
    }
```